app/modules/transcription/scripts/init.js

```diff
@@ -19,159 +19,171 @@
             });
     });
 
-    module.controller('transcriptionCtrl', function ($rootScope, $q, $timeout, $scope, $sce, $stateParams, zooAPI, zooAPISubjectSets, localStorageService, svgPanZoomFactory) {
-        
+    module.service('pendingAnnotationsService', ['zooAPI', function(zooAPI) {
+        this.get = function(subjectSet, page) {
+            // Fetch current user's incomplete classifications
+            return zooAPI.type('classifications/incomplete').get({
+                page: page || 1,
+                project_id: subjectSet.links.project
+            }).then(function(annotations) {
+                // Filter them to the current subject set
+                return Promise.all(annotations.map(function (annotation) {
+                    annotation.metadata.subjects = [];
+                    var subjectId = annotation.links.subjects[0];
+                    return zooAPI.type('set_member_subjects').get({ subject_id: subjectId })
```

Comment:
Unless I'm misunderstanding this, I don't think we need to make additional API calls to `set_member_subjects` after all. If we store `current_subject` in localStorageService (as it is in the sequential subjects PR) then we can query incomplete subjects by (parent) subject id. For example,

```
            return zooAPI.type('classifications/incomplete').get({
                page: page || 1,
                project_id: subjectSet.links.project,
                subject_id: current_subject.id
            })
```